src/contracts/core/StakeRootCompendium.sol

```diff
@@ -0,0 +1,150 @@
+        for (uint256 i = 0; i < stakeRootLeaves.length; i++) {
+            operatorSetLeaves[i] = keccak256(abi.encodePacked(stakeRootLeaves[i].avs, stakeRootLeaves[i].operatorSetRoot, stakeRootLeaves[i].operatorSetId));
+        }
+        return Merkle.merkleizeKeccak256(operatorSetLeaves);
+    }
+
+    //Note: assumes that the operator is registered to the AVS and is registered with the delegationManager.
+    function getOperatorSetRoot(
+        address avs,
+        uint32 operatorSetId, 
+        address[] calldata operators, 
+        IStrategy[] calldata strategies
+    )
+        external view 
+        isOperatorSet(avs, operatorSetId) 
+        returns (bytes32) 
+    {
+         
+        require(operators.length <= MAX_OPERATOR_SET_SIZE, "AVSSyncTree._verifyOperatorStatus: operator set too large");
+        require(operators.length == avsDirectory.operatorSetMemberCount(avs, operatorSetId), "AVSSyncTree.getOperatorSetRoot: operator set size mismatch");
+
+        bytes32[] memory operatorLeaves = new bytes32[](operators.length);
+        uint160 prevOperator = 0;
+        for (uint256 i = 0; i < operators.length; i++) {
+            require(avsDirectory.isMember(avs,operators[i], operatorSetId), "AVSSyncTree.getOperatorSetRoot: operator not in operator set");
+            require(uint160(operators[i]) > prevOperator, "AVSSyncTree.getOperatorSetRoot: operators not sorted");
+            
+            uint96[] memory multipliers = new uint96[](strategies.length);
+            for (uint256 j = 0; j < strategies.length; j++) {
+                multipliers[j] = operatorSetIdToStrategyToMultiplier[operatorSetId][strategies[i]];
+            }
+            // shares associated with this operator and these strategies
+            operatorLeaves[i] =  keccak256(abi.encodePacked(operators[i], _calculateWeightedStrategyShareSum(operators[i], strategies, multipliers)));    
+        }
+        return Merkle.merkleizeKeccak256(operatorLeaves);
+    }
+
+    function verifySnarkProof(
+        bytes calldata _journal,
+        bytes calldata _seal
+    ) external {
+        IRiscZeroVerifier(verifier).verify(
+                _seal,
+                imageId,
+                sha256(_journal)
+            );
+        
+        emit SnarkProofVerified(_journal, _seal);
+    }
+
+    //TODO: getting an error here:
+    // Error: Unimplemented feature (/solidity/libsolidity/codegen/ArrayUtils.cpp:240):
+    // Copying of type struct IStakeRootCompendium.StrategyAndMultiplier calldata[] calldata to storage not yet supported.
+    function setStrategiesAndMultipliers(
+        uint32 operatorSetId,
+        StrategyAndMultiplier[] calldata strategiesAndMultipliers
+    ) external isOperatorSet(msg.sender, operatorSetId) {
+        if(!isAVS[msg.sender]) {
+            isAVS[msg.sender] = true;
+            //if the AVS has never set its strategies and multipliers before, increment the number of AVSs
+            numAVSs++;
```

Comment:
this should be numOperatorSets not numAVSs

Reply:
did I misinterpret this comment: https://github.com/Layr-Labs/eigenlayer-contracts/pull/661#discussion_r1710064907